### PR TITLE
don't push design docs if they haven't changed

### PR DESCRIPTION
**Pull request by snopoke**
Only applies to `atomic` pushes.

Currently we push all design docs to couch regardless of whether or not there have been changes. This will allow us to only push (and index) design docs that have actually changed.

### CODE REVIEW

File: couchdbkit/designer/fs.py [hunk 1]
```diff
@@ -391,10 +410,12 @@ def push(path, dbs, atomic=True, force=False, docid=None):
         dbs = [dbs]
 
     doc = document(path, create=False, docid=docid)
-    doc.push(dbs, atomic=atomic, force=force)
+    pushed = doc.push(dbs, atomic=atomic, force=force)
     docspath = os.path.join(path, '_docs')
     if os.path.exists(docspath):
+        pushed = True
         pushdocs(docspath, dbs, atomic=atomic)
+    return pushed
```
Comment (emord) [inline couchdbkit/designer/fs.py]: will this be used anywhere?

Comment (snopoke) [reply couchdbkit/designer/fs.py]: https://github.com/dimagi/commcare-hq/compare/sk/couch-sync-skip?expand=1

File: couchdbkit/designer/fs.py [hunk 2]
```diff
@@ -102,9 +107,23 @@ def push(self, dbs, atomic=True, force=False):
...
+            if db_push:
+                logger.debug("%s/%s had been pushed from %s" % (db.uri,
+                    self.docid, self.docdir))
+            pushed |= db_push
+        return pushed
+
+    def _compare_with_current_version(self, db, doc):
+        """:returns: True if docs match otherwise False"""
+        try:
+            olddoc = db.open_doc(self._doc['_id'])
```
Comment (dannyroberts) [inline couchdbkit/designer/fs.py]: Does this mean we're going to be doing two GETs per design doc? Probably fine / still a big improvement over an unnecessary write, but just wanted to make sure I was reading this right.

Comment (snopoke) [reply couchdbkit/designer/fs.py]: That's correct. The other read happens here: https://github.com/dimagi/couchdbkit/blob/jsonobject/couchdbkit/designer/fs.py#L142 and (strangely) get's assigned to `self.olddoc`.

Technically I could just use that but it didn't feel right to make `_compare_with_current_version` rely on `self.olddoc` which gets re-assigned for each loop iteration (for each DB).

**Pull request merged**